crates/agglayer-contracts/src/lib.rs

```diff
@@ -149,31 +150,35 @@ impl<RpcProvider> L1RpcClient<RpcProvider>
 where
     RpcProvider: alloy::providers::Provider + Clone + 'static,
 {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
         rpc: Arc<RpcProvider>,
         inner: contracts::PolygonRollupManagerRpcClient<RpcProvider>,
-        l1_info_tree: Address,
+        global_exit_root_manager_contract: Address,
         default_l1_info_tree_entry: (u32, [u8; 32]),
         gas_multiplier_factor: u32,
         gas_price_params: GasPriceParams,
+        event_filter_block_range: u64,
     ) -> Self {
         Self {
             rpc,
             inner,
-            l1_info_tree,
+            global_exit_root_manager_contract,
             default_l1_info_tree_entry,
             gas_multiplier_factor,
             gas_price_params,
             l1_info_roots: Arc::new(RwLock::new(HashMap::new())),
+            event_filter_block_range,
```

Comment:
Low priority follow up suggestion: Would be nice to query the provider for max logs per request somehow, instead of fiddling with the value manually. Ideally not by parsing the error message 😄. I have no idea if the API exposes anything like that.